kratos/python/add_containers_to_python.cpp

```diff
@@ -131,6 +131,28 @@ template< class TBinderType, typename TContainerType, typename TVariableType > v
 //         }
 //     }
 
+template <class TVariableType>
+TVariableType CreateVariable(const std::string& name)
+{
+    KRATOS_ERROR_IF(KratosComponents<VariableData>::Has(name)) <<"The variable : "<<name<<" already exists."<<std::endl;
```

Comment:
We should NOT send an error if the variable exist.

We should search for variable with different name and same key as the new one then we should send an error!

Reply:
@pooyan-dadvand  I did not completely follow. So we should check if a variable with the same name and key exists ?

Reply:
@adityaghantasala In kratos it is OK to register the same variable multiple times (think from different applications) as long as all "copies" have the same type. However, a clash in key value (very unlikely, but theoretically possible) would make the new variable unusable. The safe check is to see if the key value already exists but corresponds to a variable with different name.

Reply:
ok .. I understand now .. I will make the respective changes .. !

Reply:
@jcotela @pooyan-dadvand I made the changes to check for the key also if a variable of same name exists. Can you check this ?